google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java

```diff
@@ -629,34 +620,17 @@ protected RetrySettings defaultRetrySettings() {
     return Iterables.getFirst(ServiceLoader.load(clazz), defaultInstance);
   }
 
-  private String getPomVersion() {
-    try {
-      String projectPropertiesPath = PROPERTIES_ROOT + "/"
-          + artifactId + "/" + PROPERTIES_FILE;
-      return PropertiesProvider.loadProperty(
-          ServiceOptions.class, projectPropertiesPath, PROPERTIES_VERSION_KEY);
-    } catch (Exception e) {
-      // ignore
-    }
-    return null;
+  private String getVersionProperty(String packagePath) {
+    String projectPropertiesPath = "/" + packagePath + "/" + PROPERTIES_FILE;
+    return PropertiesProvider.loadProperty(
+        ServiceOptions.class, projectPropertiesPath, PROPERTIES_VERSION_KEY);
   }
 
-  private String getManifestVersion() {
-    String version = null;
-    try {
-      Enumeration<URL> resources =
-          ServiceOptions.class.getClassLoader().getResources(JarFile.MANIFEST_NAME);
-      while (resources.hasMoreElements() && version == null) {
-        Manifest manifest = new Manifest(resources.nextElement().openStream());
-        Attributes manifestAttributes = manifest.getMainAttributes();
-        String value = manifestAttributes.getValue(MANIFEST_ARTIFACT_ID_KEY);
-        if (value != null && value.equals(artifactId)) {
-          version = manifestAttributes.getValue(MANIFEST_VERSION_KEY);
-        }
-      }
-    } catch (IOException e) {
-      // ignore
+  private String getPackagePath() {
+    List<String> fullClassName = Splitter.on(".").splitToList(this.getClass().getCanonicalName());
+    if (fullClassName.size() > 2) {
+      return String.join("/", fullClassName.subList(0, fullClassName.size() - 1));
```

Comment:
`String.join` doesn't exist in java 7 (this caused build failures)

Here is a simpler way to do this method:

```
return this.getClass().getPackage().getName().replaceAll("\\.", "/");
```